plugins/autoclothing.cpp

```diff
@@ -663,3 +682,200 @@ static void save_state(color_ostream &out)
+        else
+            out << "    " << race.second << " " << Units::getRaceNamePluralById(race.first) << endl;
+    }
+}
+
+static bool isAvailableItem(df::item* item)
+{
+    if (item->flags.bits.in_job)
+        return false;
+    if (item->flags.bits.hostile)
+        return false;
+    if (item->flags.bits.in_building)
+        return false;
+    if (item->flags.bits.in_building)
+        return false;
+    if (item->flags.bits.encased)
+        return false;
+    if (item->flags.bits.foreign)
+        return false;
+    if (item->flags.bits.trader)
+        return false;
+    if (item->flags.bits.owned)
+        return false;
+    if (item->flags.bits.artifact)
+        return false;
+    if (item->flags.bits.forbid)
+        return false;
+    if (item->flags.bits.dump)
+        return false;
+    if (item->flags.bits.on_fire)
+        return false;
+    if (item->flags.bits.melt)
+        return false;
+    if (item->flags.bits.hidden)
+        return false;
+    if (item->getWear() > 1)
+        return false;
+    if (!item->isClothing())
+        return false;
+    return true;
+}
+
+static void generate_report(color_ostream& out)
+{
+    std::map<int, int> fullUnitList;
+    std::map<int, int> missingArmor;
+    std::map<int, int> missingShoes;
+    std::map<int, int> missingHelms;
+    std::map<int, int> missingGloves;
+    std::map<int, int> missingPants;
+    for (df::unit* unit : world->units.active)
+    {
+        if (!Units::isCitizen(unit))
+            continue;
+        fullUnitList[unit->race]++;
+        int numArmor = 0, numShoes = 0, numHelms = 0, numGloves = 0, numPants = 0;
+        for (auto itemId : unit->owned_items)
+        {
+
+            auto item = Items::findItemByID(itemId);
+            if (item->getWear() >= 1)
+                continue;
+            switch (item->getType())
+            {
+            case df::item_type::ARMOR:
+                numArmor++;
+                break;
+            case df::item_type::SHOES:
+                numShoes++;
+                break;
+            case df::item_type::HELM:
+                numHelms++;
+                break;
+            case df::item_type::GLOVES:
+                numGloves++;
+                break;
+            case df::item_type::PANTS:
+                numPants++;
+                break;
+            default:
+                break;
+            }
+        }
+        if (numArmor == 0)
+            missingArmor[unit->race]++;
+        if (numShoes < 2)
+            missingShoes[unit->race]++;
+        if (numHelms == 0)
+            missingHelms[unit->race]++;
+        if (numGloves < 2)
+            missingGloves[unit->race]++;
+        if (numPants == 0)
+            missingPants[unit->race]++;
+        //out << Translation::TranslateName(Units::getVisibleName(unit)) << " has " << numArmor << " armor, " << numShoes << " shoes, " << numHelms << " helms, " << numGloves << " gloves, " << numPants << " pants" << endl;
```

Comment:
consider the Debug log for things like this. then you can call `debugfilter set debug autoclothing` at any time to turn the output on instead of editing and recompiling. See https://github.com/DFHack/dfhack/blob/develop/plugins/examples/persistent_per_save_example.cpp for an example